app/adapters/generic_adapter.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require 'uri'
+
+# KeycloakAdapter adapter to create/update/delete Clients on using the KeycloakAdapter Client Registration API.
+class GenericAdapter < AbstractAdapter
+  def self.build_client(*attrs)
+    Client.new(*attrs)
+  end
+
+  attr_reader :endpoint
+
+  def create_client(client)
+    parse http_client.put(client_url(client), body: client, header: headers)
+  end
+
+  def read_client(client)
+    parse http_client.get(client_url(client), header: headers)
+  end
+
+  def update_client(client)
+    parse http_client.put(client_url(client), body: client, header: headers)
+  end
+
+  def delete_client(client)
+    parse http_client.delete(client_url(client), header: headers)
+    client.freeze
+  end
+
+  def test
+    parse http_client.get(oidc.well_known_url, header: headers)
+  end
+
+  # The Client entity. Mapping the OpenID Connect Client Metadata representation.
+  # https://tools.ietf.org/html/rfc7591#section-2
+  class Client
+    include ActiveModel::Model
+    include ActiveModel::Conversion
+    include ActiveModel::Attributes
+
+    attr_accessor :client_id, :client_secret, :client_name, :redirect_uris, :grant_types
+
+    alias_attribute :name, :client_name
+    alias_attribute :secret, :client_secret
+    alias_attribute :id, :client_id
+
+    attr_accessor :state, :enabled
+
+    delegate :to_json, to: :to_h
+    alias read to_json
+
+    def initialize(*)
+      self.redirect_uris = []
+      self.grant_types = []
+      super
+    end
+
+    def to_h
+      {
+          client_id: client_id,
+          client_secret: client_secret,
+          client_name: client_name,
+          redirect_uris: redirect_uris,
+          grant_types: grant_types.to_ary,
+          **self.class.attributes,
+      }
+    end
+
+    def redirect_url=(val)
+      self.redirect_uris = [ val ].compact
+    end
+
+    def oidc_configuration=(config)
+      self.grant_types = GrantTypes.new(config)
```

Comment:
Wouldn't it safer/easier to assign an Array here as you already define `to_ary` in https://github.com/3scale/zync/pull/196/files#diff-9ef24b12100d60cc4d037c50e0990ae0R95?

So you do not need to call `to_ary` in https://github.com/3scale/zync/pull/196/files#diff-9ef24b12100d60cc4d037c50e0990ae0R64

Reply:
I'll try to implement `as_json` instead of `to_ary`. I want to convert it at the latest moment possible.